aws/resource_instance_profile.go

```diff
@@ -83,6 +84,20 @@ func (a InstanceProfilesAPI) Delete(instanceProfileARN string) error {
 	}, nil)
 }
 
+// Update updates the IAM role ARN of an existing instance profile
+func (a InstanceProfilesAPI) Update(ipi InstanceProfileInfo) error {
+	return a.client.Post(a.context, "/instance-profiles/edit", map[string]any{
+		"instance_profile_arn": ipi.InstanceProfileArn,
+		"iam_role_arn": func(ipi InstanceProfileInfo) string {
+			if ipi.IamRoleArn == "" {
+				return ipi.InstanceProfileArn
+			} else {
+				return ipi.IamRoleArn
+			}
+		}(ipi),
+	}, nil)
```

Comment:
This could be made much simpler:

```golang
data := map[string]any{"instance_profile_arn": ipi.InstanceProfileArn,}
if ipi.IamRoleArn != "" {
  data["iam_role_arn"]=ipi.IamRoleArn
}
return a.client.Post(a.context, "/instance-profiles/edit", data, nil)
```

Reply:
Yeah, the assignment to data is clearer, but it still needs the `else` clause.

Reply:
Modified, but left the initial assignment to `data["iam_role_arn"]` since the argument is required for the `edit` endpoint.